docs/topics/checks.txt

```diff
@@ -214,3 +214,69 @@ Messages are comparable. That allows you to easily write tests::
         )
     ]
     self.assertEqual(errors, expected_errors)
+
+
+Writing integration tests
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the need to register certain checks when the application loads,
+it can be useful to test their integration within the system checks
+framework. This can be accomplished by using
+:func:`~django.core.management.call_command`.
+
+For example, this test demonstrates that the :setting:`SITE_ID` setting must be
+an integer, a built-in
+:ref:`check from the sites framework<contrib-app-checks-sites>`::
+
+    from django.core.management import call_command
+    from django.core.management.base import SystemCheckError
+    from django.test import TestCase, modify_settings, override_settings
+
+
+    class SystemCheckIntegrationTest(TestCase):
+        @override_settings(SITE_ID="a")
+        @modify_settings(INSTALLED_APPS={"prepend": "django.contrib.sites"})
+        def test_non_integer_site_id(self):
+            regex = r"\(sites.E101\) The SITE_ID setting must be an integer."
+            with self.assertRaisesRegex(SystemCheckError, regex):
```

Comment:
Why `assertRaisesRegex` :thinking: ? and not `self.assertRaisesMessage()`
```suggestion
            msg = "(sites.E101) The SITE_ID setting must be an integer."
            with self.assertRaisesMessage(SystemCheckError, msg):
```

Reply:
🤔 I was under the impression the latter was an equality rather than a membership check. Seeing that isn't the case. Updating!

(Confession: it's been years since I've used unittest)

Reply:
Updated. This also reveals that there isn't a period at the end of the error message (and of course passed for regex).